libnm-core/nm-utils.c

```diff
@@ -4538,6 +4539,11 @@ nm_utils_inet_ntop (int addr_family, gconstpointer addr, char *dst)
 const char *
 nm_utils_inet4_ntop (in_addr_t inaddr, char *dst)
 {
+	/* relying on the static buffer (by leaving @dst as %NULL) is discouraged.
+	 * Don't do that!
+	 *
+	 * However, still support it to be lenient against mistakes and because
+	 * this is public API of libnm. */
```

Comment:
I actually liked passing NULL to use the static buffer because it saved a variable in the caller. Yes, it can lead to programming errors if there are multiple calls in the same statement... but I think it's still worth to use it

Reply:
The rest LGTM

Reply:
There are arguments for both and ultimately a matter of opinion. In the past, I thought so too, now I tend to dislike it. I find it error prone and generally ugly to use global data (when it can be avoided).